x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/TransportDeprecationCacheResetAction.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.deprecation.logging;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.logging.DeprecationCategory;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.logging.RateLimitingFilter;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+public class TransportDeprecationCacheResetAction
+    extends HandledTransportAction<DeprecationCacheResetAction.Request, ActionResponse.Empty> {
+    private static final DeprecationLogger logger = DeprecationLogger.getLogger(TransportDeprecationCacheResetAction.class);
+
+    private final RateLimitingFilter rateLimitingFilterForIndexing;
+
+    @Inject
+    public TransportDeprecationCacheResetAction( TransportService transportService, ActionFilters actionFilters,
+                                                 RateLimitingFilter rateLimitingFilterForIndexing) {
+        super(DeprecationCacheResetAction.NAME, transportService, actionFilters, DeprecationCacheResetAction.Request::new);
+        this.rateLimitingFilterForIndexing = rateLimitingFilterForIndexing;
+    }
+
+    @Override
+    protected void doExecute(Task task, DeprecationCacheResetAction.Request request, ActionListener<ActionResponse.Empty> listener) {
+        rateLimitingFilterForIndexing.reset();
+        logger.warn(DeprecationCategory.CACHE_RESET,"cache_reset", "Deprecation cache was reset");
```

Comment:
Why are we logging a deprecation warning for reseting the cache?

Reply:
I thought it would be useful to see that a reset was triggered. I chose a deprecation warning, so that you could see easily in deprecation index when the reset happened and since that moment you could see duplicated messages. (the reset affects indexing only) 
I guess you could also query the index for last "cache_reset" event and then query for the other deprecation after that event.